docs/docs/dev_docs/wallets/main.md

```diff
@@ -0,0 +1,53 @@
+# Wallets
+
+Wallets are the applications through which users manage their accounts. They also implement the interface for dapps to perform actions on behalf of a user. And in addition to these usual responsibilities present in other blockchains, wallets in Aztec also need to manage privacy keys and private state, as well as producing local proofs.
+
+In this page we will cover the main responsibilities of a wallet in the Aztec network. Refer to [_writing an account contract_](./writing_an_account_contract.md) for a tutorial on how to write a contract to back a user's account, or to [_wallet architecture](./architecture.md) for an overview of its architecture and a reference on the interface a wallet must implement.
+
+## Account setup
+
+The first step for any wallet is to let the user set up their [accounts](../../concepts/foundation/accounts/main.md). An account in Aztec is implemented by its corresponding account contract that the user must deploy to begin interacting with the network. A wallet must support at least one specific [account contract implementation](./writing_an_account_contract.md), which means being able to deploy such a contract, as well as formatting transactions for it.
+
+However, users must be able to receive funds in Aztec before deploying their account. A wallet should let a user generate a [deterministic complete address](../../concepts/foundation/accounts/keys.md#addresses-partial-addresses-and-public-keys) without having to interact with the network, so they can share with others to receive funds. This requires that the wallet pins a specific contract implementation, its initialisation arguments, a deployment salt, and a privacy key. These values yield a deterministic address, so when the account contract is actually deployed, it is available at the precalculated address. Once the account contract is deployed, the user can start sending transactions using it as the transaction origin.
+
+## Transaction lifecycle
+
+Every transaction in Aztec is broadcasted to the network as its zero-knowledge proof of correct execution, in order to preserve privacy. This means that transaction proofs are generated on the wallet and not on a remote node. This is one of the biggest differences with regard to EVM chain wallets.
+
+A wallet is responsible for first **creating** an [execution request](../../concepts/foundation/accounts/main.md#execution-requests). This means going from an intent, such as _call transfer on this contract_, to an authenticated transaction formatted for the user's account contract. As an example, given an [ECDSA-based account](https://github.com/AztecProtocol/aztec-packages/blob/95d1350b23b6205ff2a7d3de41a37e0bc9ee7640/yarn-project/noir-contracts/src/contracts/ecdsa_account_contract/src/main.nr#L1), an execution request has the account contract as `origin`, calls its `entrypoint` function, and encodes the user intent as `payload`, which is signed using ECDSA with a private key managed by the wallet.
```

Comment:
> A wallet is responsible for first **creating** an [execution request](../../concepts/foundation/accounts/main.md#execution-requests). This means going from an intent, such as _call transfer on this contract_, to an authenticated transaction formatted for the user's account contract

So does a wallet _include_ an implementation of 'aztec.js', or is a wallet just the Aztec RPC Server?
I guess I'm confused still by what an intent is, in this context. A human thought (an intent such as "call transfer on this contract") is translated into the human pressing a button, which is translated by aztec.js into a JSON RPC call to the Aztec RPC Server, which is translated into arguments that can be consumed by an account contract.

Would it be truer to say that the wallet's responsibilities actually begin when it receives a JSON RPC transaction request object, and then the wallet takes that object and formats it into a 'transaction request' format that can be read by the simulator (which then formats the transaction request again into arguments that can be read by the account contract)? (I.e. that the translation of an intent into JSON RPC format is actually done outside of the wallet, before reaching the wallet?)

Reply:
Perhaps a takeaway from some of my questions so far, is that we'll need to be consistent about naming things, perhaps to the point of eventually needing a glossary, or links to actual javascript object descriptions.

Some terms that have come up so far over the past couple of pages (and in my brain):

Intent, transaction, authenticated transaction, execution request, transaction request, payload, call, transaction origin, account, account contract, account contract implementation, the various formattings of a 'transaction request' for various components, function call, enqueued function, function request, entrypoint, transaction formatting, transaction authentication, execution trace, side effect.